File: java/bitcoin/Number/NumberFactory1.java
import java.math.BigInteger;
import java.util.Random;

public class NumberFactory1 extends NumberFactory
{
  @Override public Number zero(){ return Number1.ZERO; }
  @Override public Number one(){ return Number1.ONE; }
  @Override public Number fromBytes(int signum, byte[] val){
    return new Number1(new BigInteger(signum, val));}
  @Override public Number random(int numBits, Random rnd){
    return new Number1(new BigInteger(numBits, rnd));}
}

File: c#/bitcoin/Number/NumberFactory1.cs
using System;
using System.Numerics;
using System.Security.Cryptography;
using Random = System.Security.Cryptography.RandomNumberGenerator;

public class NumberFactory1 : NumberFactory
{
  public override Number Zero(){ return Number1.ZERO; }
  public override Number One(){ return Number1.ONE; }
  public override Number FromBytes(int sign, byte[] val){  // big-endian
    return Number1.FromBytes(sign, val); }
  public override Number Random(int numBits, Random rnd){
    return Number1.Random(numBits, rnd); }
  public override Number FromBigInteger(BigInteger n){
    return new Number1(n);  }
}
